pkg/retry/retry.go

```diff
@@ -96,6 +96,11 @@ func isRetryable(err error) bool {
 			}
 		}
 		return true
+	case net.Error:
+		if e.Timeout() {
+			return true
+		}
+		return isRetryable(e)
```

Comment:
Isn’t this an infinite loop?

A fallthrough doesn’t scale, but should work fine.

Reply:
fallthrough is not allowed in this type of switch.

Oops wrong function should have been
return  isErrnoRetryable(e)

net.Error is not unwappabile.

Reply:
- `isErrnoRetryable` only ever returns `true` for `syscall.Errno`, which was checked earlier in the `switch`, so AFAICS it always return `false` here.
- Like `isErrnoRetryable` demonstrates (on `syscall.Errno` values), a single Go type can _both_ implement `net.Error` and some other interface like `unwrapper`. See the earlier-linked https://github.com/golang/go/blob/52f68efa45a34e60e8fc5a2ad5fc124a865ac2a4/src/crypto/tls/conn.go#L186-L187 for at least one example.